gamemodes/terrortown/gamemode/server/sv_armor.lua

```diff
@@ -0,0 +1,184 @@
+
+	net.Start("ttt2_sync_armor")
+	net.WriteUInt(math.Round(self.armor), 16)
+	net.Send(self)
+end
+
+---
+-- Sets the max @{ARMOR} value to a specific value.
+-- @param number armor_max The new max armor to be set
+-- @realm server
+function plymeta:SetMaxArmor(armor_max)
+	self.armor_max = math.max(armor_max, 0)
+
+	net.Start("ttt2_sync_armor_max")
+	net.WriteUInt(math.Round(armor_max), 16)
+	net.Send(self)
+
+	-- make sure armor is always smaller than the max armor
+	self:SetArmor(math.min(self:GetMaxArmor(), self:GetArmor()))
+end
+
+---
+-- Increases the @{ARMOR} directly about a specific value, while also increasing the maximum
+-- armor value. Mostly used when an armor item is given.
+-- @param number armor The amount to be increased
+-- @realm server
+function plymeta:GiveArmor(armor)
+	self:IncreaseMaxArmorValue(armor)
+	self:IncreaseArmorValue(armor)
+end
+
+---
+-- Decreases the @{ARMOR} about a scaled specific value depending on the current max value,
+-- while the maximum armor value gets decreased too.
+-- @param number remove The amount to be decreased
+-- @realm server
+function plymeta:RemoveArmor(armor)
+	self:DecreaseMaxArmorValue(armor)
+end
+
+---
+-- Decreases the @{ARMOR} directly about a specific value while keeping the max value.
+-- To remove a player's armor(item), @{plymeta:RemoveArmor()} should be used!
+-- @param number armor The amount to be decreased
+-- @realm server
+function plymeta:DecreaseArmorValue(armor)
+	self:SetArmor(self:GetArmor() - math.max(armor, 0))
+end
+
+---
+-- Increases the @{ARMOR} directly about a specific value while keeping the max value.
+-- To add a player's armor(item), @{plymeta:GiveArmor()} should be used!
+-- @param number armor The amount to be increased
+-- @realm server
+function plymeta:IncreaseArmorValue(armor)
+	self:SetArmor(self:GetArmor() + math.max(armor, 0))
+end
+
+---
+-- Decreases the maximum @{ARMOR} directly about a specific value.
+-- @param number armor The amount to be decreased
+-- @realm server
+function plymeta:DecreaseMaxArmorValue(armor)
+	self:SetMaxArmor(self:GetMaxArmor() - math.max(armor, 0))
+end
+
+---
+-- Increases the maximum @{ARMOR} directly about a specific value.
+-- @param number armor The amount to be increased
+-- @realm server
+function plymeta:IncreaseMaxArmorValue(armor)
+	self:SetMaxArmor(self:GetMaxArmor() + math.max(armor, 0))
+end
+
+---
+-- Resets the @{ARMOR} value including the max value, called in @{GM:PlayerSpawn}.
+-- @realm server
+-- @internal
+function plymeta:ResetArmor()
+	self:SetArmor(0)
+	self:SetMaxArmor(0)
+end
+
+---
+-- Sets the initial @{Player} Armor, called in @{GM:TTTBeginRound} after (dumb) players
+-- are respawned. Therefore no armor has to be reset and it is safe that the player
+-- receives their armor.
+-- @realm server
+-- @internal
+function ARMOR:InitPlayerArmor()
+	for _, p in ipairs(player.GetAll()) do
+		if p:IsTerror() then
+			p:GiveArmor(self.cv.armor_on_spawn:GetInt())
```

Comment:
GiveArmor, not SetArmor?

Reply:
yes, since GiveArmor also handles the increase of MaxArmor, SetArmor does not increase the bound and is therefore capped at maxarmor